packages/features/form-builder/FormBuilderField.tsx

```diff
@@ -53,9 +54,20 @@ export const FormBuilderField = ({
 }) => {
   const { t } = useLocale();
   const { control, formState } = useFormContext();
+  const searchParams = useRouterQuery();
 
   const { hidden, placeholder, label } = getAndUpdateNormalizedValues(field, t);
 
+  const isPreFilledInUrl = (): boolean => {
+    const message = (formState?.errors?.responses?.message || "") as string;
+    const name = message.replace(/\{([^}]+)\}.*/, "$1");
+    //if the field is already prefilled in url and it has error then we don't want to disable it
+    if (name === field.name) {
+      return false;
+    }
+    return !!(field.disableOnPrefill && searchParams && searchParams[field.name]);
```

Comment:
Could be simplified by doing early return at the top with the following code

```
if (field.disableOnPrefill || !searchParams) {return false}
```
After that it can be simply 
```suggestion
    return !!searchParams[field.name]);
```